core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -62,19 +62,22 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
 
   def resizeThreadPool(newSize: Int): Unit = {
     def migratePartitions(newSize: Int): Unit = {
+      val allRemovedPartitionsMap = mutable.Map[TopicPartition, InitialFetchState]()
       fetcherThreadMap.forKeyValue { (id, thread) =>
-        val partitionStates = removeFetcherForPartitions(thread.partitions)
+        val partitionStates = thread.removeAllPartitions()
+        failedPartitions.removeAll(partitionStates.keySet)
```

Comment:
As we end up removing all partitions from all the fetchers, it seems to me that we could just clear the whole set of failed partitions right before calling `addFetcherForPartitions(allRemovedPartitionsMap)`. We would need to add a `clear` method to `FailedPartitions`. Is my reasoning correct?

Reply:
I am not very  familiar with `FailedPartitions`, and I find that 'FailedPartitions' removed when add topic partitions, remove partitions of fetchers. I think maybe the union of `FailedPartitions` and `removedFetcherForPartitions` is the whole set of partitions. So should we just clear it or remain those topic partitions failed.

Reply:
Removing `failedPartitions.removeAll(partitionStates.keySet)` seems fine as we remove them anyway in `addPartitions`. Would it be possible to extend the test to include a few failed partitions and verify that they are indeed removed when the resizing is completed? We might be able to use `addFailedPartition`.

Reply:
done